### PR TITLE
chore: update to go 1.19 and other library updates

### PR DESCRIPTION
**Pull request by ajayk**
- updates to go 1.19 
- updates github actions 
- updates golang dependencies 
- switched to yaml.v3

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ func ReadConfig(path string, readFn ReadFileFn) (*Config, error) {
 		return nil, err
 	}
 	config := &Config{}
-	if err := yaml.UnmarshalStrict(data, config); err != nil {
+	if err := yaml.Unmarshal(data, config); err != nil {
```
Comment (mikouaj) [inline internal/config/config.go:154]: We would like to preserve strict unmarshalling behaviour in yaml.3 as well.
This is possible via `Decoder.KnownFields`, can you please reflect on this ?

From [API v3 of the yaml package for Go is available](https://ubuntu.com/blog/api-v3-of-the-yaml-package-for-go-is-available):
> The general “strict” term is gone from the API, and has been replaced by the more specific Decoder.KnownFields method which forces the decoder to ensure that any mapping keys being decoded from the YAML text are present in the struct being decoded into.

**Review by mikouaj — COMMENTED** [review]
Thank you for opening your first pull request in this repository!
Bumping dep versions is a thing to do, good job.
I've left minor comment about YAML strict unmarshalling, please take a look.

File: internal/config/config.go (hunk 1 above)
Comment (ajayk) [reply internal/config/config.go]: @mikouaj : Thanks  ,  addressed the review comments

**Review by mikouaj — APPROVED** [review]
LGTM!

**Pull request merged**